File: 0038-count-and-say/0038-count-and-say.cpp
class Solution {
public:
    string findfreq(string s){
        int count=1;
        string s2;
        for(int i=1; i<s.size();i++){
                if(s[i]==s[i-1]) {
                    count++;
                }
                else{
                    s2=s2+to_string(count)+s[i-1];
                    count=1;
                }
            }
         s2=s2+to_string(count)+s[s.size()-1];
        return s2;
    }
    string countAndSay(int n) {
      
        if(n==1) return "1";
          string s="1";
        for(int i=1; i<n;i++){
            s=findfreq(s);
        }
        return s;
    }
};

File: 0038-count-and-say/0038-count-and-say.cs
public class Solution
{
    public string CountAndSay(int n)
    {
        if (n == 1) return "1";

        string result = "1";
        for (int i = 1; i < n; i++)
        {
            result = GetNextSequence(result);
        }
        return result;
    }

    private string GetNextSequence(string s)
    {
        var sb = new System.Text.StringBuilder();
        int count = 1;

        for (int i = 1; i < s.Length; i++)
        {
            if (s[i] == s[i - 1])
            {
                count++;
            }
            else
            {
                sb.Append(count).Append(s[i - 1]);
                count = 1;
            }
        }

        // Append the last counted character
        sb.Append(count).Append(s[s.Length - 1]);

        return sb.ToString();
    }
}
